### ISSUE TITLE
Fail to build with --isolate-install and multiple source dependencies

### ISSUE THREAD

**Issue by miguelprada** [catkin/catkin_tools#451]
### System Info

* Operating System: `Linux mibuntu 3.13.0-108-generic #155-Ubuntu SMP Wed Jan 11 16:58:52 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux`
* Python Version: `Python 2.7.6`
* Version of catkin_tools: `catkin_tools 0.4.4 ...`
* ROS Distro: `indigo`

### Build / Run Issue

I was experimenting with the different options in `catkin build`, specifically with the `--isolate-install` (I was exploring whether it could be used to create some quick and dirty debs using [fpm](https://github.com/jordansissel/fpm)), and it turns out that my workspace was refusing to build with the isolated install option.

The simplest case I was able to find where the issue arises is:

- three packages in the workspace
- `pakcage_a` and `package_b` with no dependencies
- `package_c` depending on both `package_a` and `package_b`

This workspace fails to build when the `--isolate-install` option is used. My guess is that when the environment for building `package_c` is prepared, the environment setup for `package_a` is overwritten by the environment setup for `package_b`.

It's probably quite obvious from the issue description, but just for completeness sake :wink:...

* [x] Works with `catkin_make`
* [x] Works with `catkin_make_isolated --merge`
* [x] Works with `catkin build`
* [x] Works with `catkin build -p1`
* [ ] I did  not `read this`

### Expected Behavior

All three packages are built and installed into three separate FHS sub-trees (from where hopefully debs could be quickly generated with fpm).

### Actual Behavior

`find_package( catkin REQUIRED COMPONENTS package_a package_b )` fails to find configuration file produced by `package_a` and the build fails.

You can see the output of the (verbose) `catkin build` [here](https://gist.github.com/miguelprada/5f94a1b678e59e62588db8040255e51a), and the commands used to reproduce below.

### Steps to Reproduce the Issue

```bash
mkdir -p /tmp/test_ws/src
cd /tmp/test_ws/src
catkin create pkg package_a
catkin create pkg package_b
catkin create pkg package_c --catkin-deps package_a package_b
cd /tmp/test_ws
catkin config --extend /opt/ros/indigo --install --isolate-install
VERBOSE=1 catkin build -p1 -v
```

**Comment by mikepurvis** [comment]
This is the exact scenario discussed in https://github.com/catkin/catkin_tools/issues/382#issuecomment-285739921 and which was fixed by #443 and its follow-on #449.

Please try the branch from #449 and see if that works for your use case.

**Comment by miguelprada** [comment]
Sigh... I searched for similar issues before creating a new one, but clearly failed to do so properly. My apologies.

I tried to test #449, but I'm unsure of how to use a source-based version of `catkin_tools` without messing up my `apt-get` based installation (tried with `virtualenv`, but something failed and I'm not confident enough to move forward).

Since this is not critical for me right now, I'll just wait for that patch to be released and test it then. I'll comment when that happens with confirmation. Thanks, @mikepurvis!
